programs/mango-v4/src/accounts_ix/openbook_v2_liq_force_cancel_orders.rs

```diff
@@ -0,0 +1,69 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{Token, TokenAccount};
+
+use crate::error::*;
+use crate::state::*;
+use openbook_v2::{
+    program::OpenbookV2,
+    state::{BookSide, Market, OpenOrdersAccount},
+};
+
+#[derive(Accounts)]
+pub struct OpenbookV2LiqForceCancelOrders<'info> {
+    #[account(
+        constraint = group.load()?.is_ix_enabled(IxGate::OpenbookV2LiqForceCancelOrders) @ MangoError::IxIsDisabled,
+    )]
+    pub group: AccountLoader<'info, Group>,
+
+    // Allow force cancel even if account is frozen
+    #[account(
+        mut,
+        has_one = group
+    )]
+    pub account: AccountLoader<'info, MangoAccountFixed>,
+
+    #[account(mut)]
+    pub open_orders: AccountLoader<'info, OpenOrdersAccount>,
+
+    #[account(
+        has_one = group,
+        has_one = openbook_v2_program,
+        has_one = openbook_v2_market_external,
+    )]
+    pub openbook_v2_market: AccountLoader<'info, OpenbookV2Market>,
+
+    pub openbook_v2_program: Program<'info, OpenbookV2>,
+
+    #[account(mut)]
+    pub openbook_v2_market_external: AccountLoader<'info, Market>,
+
+    #[account(mut)]
+    pub market_bids: AccountLoader<'info, BookSide>,
```

Comment:
again, validation comments would be good. Also, I don't thing we really need to use `AccountLoader<'info, BookSide>` right? we never interact with the data there directly, so could just be an UncheckedAccount that we forward to the openbook program instruction?

Reply:
adding some constaints.